src/api/app/controllers/webui/project_controller.rb

```diff
@@ -471,4 +476,13 @@ def set_project_by_name
   rescue Project::UnknownObjectError
     @project = nil
   end
+
+  def report_bug_or_bugzilla_url
```

Comment:
This smells like a controller concern to me that is shared between `Package` and `Project`.

Reply:
Yes. In this pull request I extracted some logic from the views to the controller. I preferred not to provide an over-refactoring. We could keep refactoring this code later on.

Reply:
Not sure how this would qualify as `over-refactoring`. Understanding this code is now quite some gymnastics between view, helper and controller. And what better point in time to refactor than when introducing a major new feature?

Reply:
Hm, I even think this is model code (in a concern).

Reply:
I moved the `report_bug_or_bugzilla_url` method into a model concern, as you suggested. Please take a look again.